browser/ui/brave_tab_strip_model_delegate.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_BRAVE_TAB_STRIP_MODEL_DELEGATE_H_
+#define BRAVE_BROWSER_UI_BRAVE_TAB_STRIP_MODEL_DELEGATE_H_
+
+#include <memory>
+#include <vector>
+
+#include "chrome/browser/ui/browser_tab_strip_model_delegate.h"
+
+// In order to make it easy to replace BrowserTabStripModelDelegate with our
+// BraveTabStripModelDelegate, wrap this class with chrome namespace.
+namespace chrome {
+
+class BraveTabStripModelDelegate : public BrowserTabStripModelDelegate {
+ public:
+  using BrowserTabStripModelDelegate::BrowserTabStripModelDelegate;
+  BraveTabStripModelDelegate(const BraveTabStripModelDelegate&) = delete;
+  BraveTabStripModelDelegate& operator=(const BraveTabStripModelDelegate&) =
+      delete;
+  BraveTabStripModelDelegate(BraveTabStripModelDelegate&&) noexcept = delete;
+  BraveTabStripModelDelegate& operator=(BraveTabStripModelDelegate&&) noexcept =
+      delete;
```

Comment:
This one too? The base class deletes copy ctors already, so we could omit them too.

Reply:
Ah, I was wrong. Below seems correct.
```
Move operators will not be generated when copy operations are declared explicitely.
```
In chromium project, most of classes delete copy operators explicitely to prevent copying instance.
However, seems allowing implicit move operator generation. So, I think it's up to you for deleting move operators explicitely like above.

Reply:
> Move operators will not be generated when copy operations are declared explicitely.

I believe `=delete` is also considered as user-declared copy constructor, which means compiler doesn't generate move ops implicitly - I think you were right.

Reply:
I'm reading `Effective modern c++` now :)
According to that book, move operators are only generated when class doesn't have user-declared copy, move or destructors. So, I think `BraveTabStripModelDelegate` doesn't need to delete explicitely move operators because it has user declared dtor. and also doesn't need to delete copy operators as base class already deleted. does it correct? :)

Reply:
Yes, that's what I understand. It seems that defaulting/deleting constructor from one base class is more preferable and, when we default/delete one of them, it'd be better to do the same for all of them (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all)